saveEventsFile.m

```diff
@@ -59,86 +41,89 @@
     % to true then this will tell you where the file is located.
     %
     % See test_saveEventsFile in the test folder for more details on how to use it.
-    
+
+    if nargin < 1
+        errror('Missing action input');
+    end
+
     if nargin < 3 || isempty(logFile)
         logFile = struct();
     end
-    
+
     switch action
-        
+
         case 'open'
-            
+
             logFile.filename = expParameters.fileName.events;
-            
-            logFile = initializeFile(expParameters, logFile, varargin);
-            
+
+            logFile = initializeFile(expParameters, logFile);
+
         case 'open_stim'
-            
+
             logFile.filename = expParameters.fileName.stim;
-            
-            logFile = initializeFile(expParameters, logFile, varargin);
-            
+
+            logFile = initializeFile(expParameters, logFile);
+
         case 'save'
-            
+
             if ~isstruct(logFile) || size(logFile, 2) > 1
                 error('The variable containing the n events to save must be a nx1 structure.');
             end
-            
+
             % appends to the logfile all the data stored in the structure
             % first with the standard BIDS data and then any extra things
             for iEvent = 1:size(logFile, 1)
-                
+
                 onset = checkInput(logFile(iEvent).onset);
                 duration = checkInput(logFile(iEvent).duration);
                 trial_type = checkInput(logFile(iEvent).trial_type);
-                
-                fprintf(logFile(1).fileID, '%f\t%s\t%f\t', ...
-                    onset, ...
-                    trial_type, ...
-                    duration);
-                
-                for iExtraColumn = 1:numel(varargin)
-                    
-                    % if the field we are looking for does not exist or is empty in the
-                    % action logFile structure we will write a NaN otherwise we
-                    % write its content
-                    
-                    data = 'NA';
-                    if isfield(logFile, varargin{iExtraColumn})
-                        data = getfield(logFile(iEvent), varargin{iExtraColumn}); %#ok<GFLD>
-                    end
-                    
-                    data = checkInput(data);
-                    
-                    if ischar(data)
-                        fprintf(logFile(1).fileID, '%s\t', data);
-                    else
-                        fprintf(logFile(1).fileID, '%f\t', data);
-                    end
-                    
+
+                if any(isnan([onset trial_type])) || ...
+                        any(isempty([onset trial_type])) || ...
+                        any(strcmp({onset, trial_type}, 'NA'))
+
+                    warning('\nSkipping saving this event.\n onset: %f \n trial_type: %s\n', ...
+                        onset, ...
+                        trial_type);
+
```

Comment:
I didn't have time to check README file yet, but do we want to indicate there **again** that if the user DOES NOT provide `onset`, `trial_type`, this events will be skipped? (and/or for BIDS these are compulsory)
Or is it OK to have the warning as you did at the beginning of this function? 
I'm OK with either option :) it's BIDS repo, people should know a bit about what bids asks for.

Second thing is, `duration` can be skipped?

Reply:
So right now duration will be set to "NaN" if it is not set. I need to double check if this is BIDS friendly. I suspect it is not. Not sure.

You are correct though that this might need to be mentioned in the README: people might get confused otherwise.

Reply:
Another thing I could do is set the default for duration to 0. In the SPM world this would turn this into a "pure" event related trial.